src/connection.rs

```diff
@@ -283,11 +283,7 @@ impl ActualConnection {
     pub fn send_bytes(&mut self, bytes: &[u8]) -> RedisResult<Value> {
         match *self {
             ActualConnection::Tcp(ref mut connection) => {
-                let res = connection
-                    .reader
-                    .get_mut()
-                    .write_all(bytes)
-                    .map_err(RedisError::from);
+                let res = connection.reader.write_all(bytes).map_err(RedisError::from);
```

Comment:
Why did we name this `reader` when we're clearly writing to it?
(I know it wasn't part of this PR, I just stumbled upon it here)

Reply:
Ah well, it was a `Buf_Reader_` previously. Let's not tack this on this PR though, I'm happy to do the search-and-replace afterwards.